### DISCUSSION TITLE
How to stream documents that match a query?

### DISCUSSION THREAD

**Discussion by mathew-kurian** [invertase/react-native-firebase#4339] [Q&A] [answered]
I see a Stream API on the NodeJS side https://googleapis.dev/nodejs/firestore/latest/CollectionReference.html#stream. Does the same exist on the react native end?

**Comment by mikehardy** [comment, marked as answer]
It does not, the native APIs on android/ios aren't present: https://firebase.google.com/docs/reference/android/com/google/firebase/firestore/CollectionReference - in fact the link you reference is for the admin sdk only, it's not even present on the firebase-js-sdk

I have needed something like this in my project. I implemented a cloud function that used the admin sdk to do the work and return the needful

**Reply by mathew-kurian** [reply]
@mikehardy Thank you. That's a good idea. Are there any restrictions/considerations to keep in mind with streaming responses on the native devices via. functions?

**Reply by mikehardy** [reply]
well, you won't be streaming to the device at all, the device will be waiting until you've finished processing completely in the function and returned a response, so you may have higher latency if you are streaming one big stream and processing before returning. In my case I was streaming multiple things at once and doing an awaitAll which is pretty quick in parallel with functions GCP VMs being colocated with firestore nodes for me, so latency to display the result on the screen was decreased. It's very very requirement specific
